core/src/main/java/com/datastax/oss/driver/internal/core/CqlIdentifiers.java

```diff
@@ -16,21 +16,36 @@
 package com.datastax.oss.driver.internal.core;
 
 import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 
 public class CqlIdentifiers {
 
-  public static List<CqlIdentifier> wrap(Iterable<String> in) {
+  public static List<CqlIdentifier> wrap(
+      @NonNull Iterable<String> in, @NonNull Function<String, CqlIdentifier> fn) {
+
+    Preconditions.checkNotNull(in, "Input Iterable must not be null");
+    Preconditions.checkNotNull(fn, "CqlIdentifier conversion function must not be null");
     ImmutableList.Builder<CqlIdentifier> out = ImmutableList.builder();
     for (String name : in) {
-      out.add(CqlIdentifier.fromCql(name));
+      out.add(fn.apply(name));
     }
     return out.build();
   }
 
+  public static List<CqlIdentifier> wrap(Iterable<String> in) {
+    return wrap(in, CqlIdentifier::fromCql);
+  }
+
+  public static List<CqlIdentifier> wrapInternal(Iterable<String> in) {
+    return wrap(in, CqlIdentifier::fromInternal);
+  }
```

Comment:
I do wonder about a couple things here:

* Presumably we should annotate the args to these methods with @NonNull as well?  Not sure if any static checker is smart enough to realize this method is a straight cut-through and that the params of the caller should be applied here
* Presumably both of these methods (as well as the underlying wrap(Iterable, Fn) call) should also be annotated with @NonNull?

Reply:
Agreed, I'll add those in the next push

Reply:
We didn't have the nullability annotations initially. We've added them all over the public API but we yet have to do a pass on the internal packages. When we add new internal methods I agree that it doesn't hurt to have the annotations from the start.